Mathlib/Combinatorics/Enumerative/DyckWord.lean

```diff
@@ -196,4 +197,139 @@ lemma two_mul_semilength_eq_length : 2 * p.semilength = p.toList.length := by
 
 end Semilength
 
+section FirstReturn
+
+/-- `p.firstReturn` is 0 if `p = 0` and the index of the `D` matching the initial `U` otherwise. -/
+def firstReturn : ℕ :=
+  (range p.toList.length).findIdx (fun i ↦
+    (p.toList.take (i + 1)).count U = (p.toList.take (i + 1)).count D)
+
+@[simp] lemma firstReturn_zero : firstReturn 0 = 0 := rfl
+
+lemma firstReturn_pos (h : p.toList ≠ []) : 0 < p.firstReturn := by
```

Comment:
Once you have applied the other comment, this lemma will be in a form that positivity understands. Could you write the corresponding positivity extension? Feel free to cargo-cult an existing one

Reply:
In order to write such an extension I need a partial order on Dyck words. There appears to be no suitable such partial order.